service/src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -114,6 +115,23 @@ public AzureApi samAzureApi(String accessToken) {
     return new AzureApi(getApiClient(accessToken));
   }
 
+  public AdminApi samAdminApi(String accessToken) {
+    return new AdminApi(getApiClient(accessToken));
+  }
+
+  @Traced
+  public boolean isAdmin(AuthenticatedUserRequest userRequest) throws InterruptedException {
+    try {
```

Comment:
@zloery it's unclear to me how to check the verilygroup access. the api is not exposed. https://g3doc.corp.google.com/company/teams/verily/platform_eng/verilygroups/guides/api_guide.md?cl=head

i'd like your early thought on just querying the sam admin api to check for admin access..

Reply:
I actually didn't realize they exposed that much of the API - the RPC endpoints aren't exposed but it looks like the HTTP ones are, and `GetMember` seems like the method we're looking for. That would be a pretty clean solution within Verily (especially because it would let us validate breakglass access). It does seem like it wouldn't mesh well with the Broad's admin approach, though.

Does Sam have an endpoint for checking admin access? Generally I think that's granted via membership in the `fc-admins` group but I'm not sure if that's a Sam group or just a googlegroup. If there's something like that available that SGTM

Reply:
> I actually didn't realize they exposed that much of the API - the RPC endpoints aren't exposed but it looks like the HTTP ones are, and `GetMember` seems like the method we're looking for. That would be a pretty clean solution within Verily (especially because it would let us validate breakglass access). It does seem like it wouldn't mesh well with the Broad's admin approach, though.

where do you see the http endpoint? yeah we can flag guard it so we need breakglass access for verily env and use sam for broad.
> 
> Does Sam have an endpoint for checking admin access? Generally I think that's granted via membership in the `fc-admins` group but I'm not sure if that's a Sam group or just a googlegroup. If there's something like that available that SGTM

i didn't see a sam endpoint to directly verify if a user is admin. so I'm just calling a sam AdminApi and see if I have permissions...https://github.com/broadinstitute/sam/blob/03ffedde48a9fd810cc1086429f60675064310b4/src/test/scala/org/broadinstitute/dsde/workbench/sam/api/AdminUserRoutesSpec.scala#L32

Reply:
That page links to their published API (though all their line number links are broken 😬). In this case, the endpoint is `GET /v1/groups/{group_id}/members/{member_id}`. They also have some example code for dealing with IAP, though I haven't done that before.

I would double check with the Identiteam before building something like that or if they have a recommended way to check admin status. It seems like it should work, but it's definitely an unexpected way to use that Sam endpoint

Reply:
confirmed with Broad (dvoet) that this is an OK way of checking admin. calling this api check for terra level admin (sam has some different admin types but this is the one we needed)

i can pass through userRequest through iamCow in the flight that effectively check for breakglass. But that makes testing impossible as the test user is not admin.